lib/moonshot/stack.rb

```diff
@@ -154,12 +155,15 @@ def default_values
     end
 
     def template
-      @template ||= StackTemplate.new(template_file, log: @log)
+      @template ||= load_template_file
     end
 
     # @return [String] the path to the template file.
     def template_file
-      File.join(Dir.pwd, 'cloud_formation', "#{@app_name}.json")
+      json = "#{raw_template_file_name}.json"
+      yaml = "#{raw_template_file_name}.yml"
+      @template_file ||= Dir[json].first || Dir[yaml].first || \
+                         raise('No conforming template file was found.')
```

Comment:
I like to always give the user full context so they're not confused about what is expected. What about:
```ruby
@template_file ||= Dir[json].first || Dir[yaml].first
raise "CloudFormation template not found at #{json} or #{yaml}!" unless @template_file
```